### PR TITLE
ENT-1463 Course duration in view from data consent screen

### PR DESCRIPTION
**Pull request by irfanuddinahmad**
**Description:** This PR adds course duration to the view from data consent screen.

**JIRA:** https://openedx.atlassian.net/browse/ENT-1463

Test Instructions:

1: Enroll in a course e.g. http://edx.devstack.lms:18000/enterprise/1a846f05-42bf-4cfe-91bf-30bbb937ffbb/course/course-v1:edX+DemoX+2019-T1/enroll/
2: Click Course Details in 'Confirm Your Course' page
3: The new field should appear as:
course-duration

### CODE REVIEW

File: enterprise/views.py [hunk 1]
```diff
@@ -1022,6 +1032,7 @@ def get_enterprise_course_enrollment_page(
             'continue_link_text': _('Continue'),
             'level_text': _('Level'),
             'effort_text': _('Effort'),
+            'duration_text': _('Duration'),
```
Comment (mushtaqak) [inline enterprise/views.py]: suggestion: `duration_title`

File: enterprise/templates/enterprise/templatetags/course_modal.html [hunk 2]
```diff
@@ -73,6 +73,19 @@ <h1 id="modal-header-text-{{ index }}" class="modal-header-text">{{ course_title
                 </div>
               </li>
             {% endif %}
+            {% if course_duration %}
+              <li>
+                <div class="detail-container">
+                  <div class="detail-title-container">
+                    <span class="icon fa fa-tachometer" aria-hidden="true"></span>
```
Comment (mushtaqak) [inline enterprise/templates/enterprise/templatetags/course_modal.html]: Should we use a different icon? may be `fa-clock`?

File: enterprise/views.py [hunk 3]
```diff
@@ -950,6 +951,14 @@ def get_enterprise_course_enrollment_page(
                 course_start_date = parse(course_run['start']).strftime('%B %d, %Y')
                 now = datetime.datetime.now(pytz.UTC)
                 course_in_future = parse(course_run['start']) > now
+            if course_run['start'] and course_run['end'] and course_run['weeks_to_complete']:
+                course_end_date = parse(course_run['end']).strftime('%B %d, %Y')
+                weeks_to_complete = course_run['weeks_to_complete']
+                course_duration = _('{num_weeks} weeks, Starting on {start_date} and ending at {end_date}').format(
+                    num_weeks=weeks_to_complete,
```
Comment (mushtaqak) [inline enterprise/views.py]: I think we can format `weeks_to_complete` so that it handles singular and plural weeks.
```
total_weeks=ungettext(
    '{} week',
    '{} weeks',
    weeks_to_complete
).format(weeks_to_complete) if weeks_to_complete else ''
```

File: tests/test_enterprise/views/test_course_enrollment_view.py [hunk 4]
```diff
@@ -644,7 +644,7 @@ def test_get_course_enrollment_page_course_unenrollable(
         The message indicating that the course is currently unopen to new learners is rendered.
         """
         setup_course_catalog_api_client_mock(course_catalog_client_mock, course_run_overrides={
-            'end': None if enrollable else '1000-10-13T13:11:01Z',
+            'end': None if enrollable else '1900-10-13T13:11:01Z',
```
Comment (mushtaqak) [inline tests/test_enterprise/views/test_course_enrollment_view.py]: why was this changed?

File: enterprise/views.py (hunk 1 above)
Comment (irfanuddinahmad) [reply enterprise/views.py]: Following the same pattern as for effort_text

File: tests/test_enterprise/views/test_course_enrollment_view.py (hunk 4 above)
Comment (irfanuddinahmad) [reply tests/test_enterprise/views/test_course_enrollment_view.py]: strftime requires dates after 1900

File: enterprise/views.py (hunk 3 above)
Comment (irfanuddinahmad) [reply enterprise/views.py]: Done

File: enterprise/templates/enterprise/templatetags/course_modal.html (hunk 2 above)
Comment (irfanuddinahmad) [reply enterprise/templates/enterprise/templatetags/course_modal.html]: Done
